share/lib/python/neuron/gui.py

```diff
@@ -86,30 +86,58 @@ def end(self):
 
 class LoopTimer(threading.Thread):
     """
-    a Timer that calls f every interval
+    A Timer that calls a function at regular intervals.
     """
 
     def __init__(self, interval, fun):
-        """
-        @param interval: time in seconds between call to fun()
-        @param fun: the function to call on timer update
-        """
         self.started = False
         self.interval = interval
         self.fun = fun
+        self._running = threading.Event()
         threading.Thread.__init__(self, daemon=True)
 
     def run(self):
         h.nrniv_bind_thread(threading.current_thread().ident)
         self.started = True
-        while True:
+        self._running.set()
+        while self._running.is_set():
             self.fun()
             time.sleep(self.interval)
 
+    def stop(self):
+        """Stop the timer thread and wait for it to terminate."""
+        self._running.clear()
+        self.join()
+
+
+if h.nrnversion(9) == "2":  # Launched with Python (instead of nrniv)
+    from neuron.exec_in_main_thread import submit_task_to_main_thread
```

Comment:
I suggest that 1bd875120 be reverted (will remove exec_in_main_thread.py). It does fix the thread sanitizer leak when exiting with NEURONMainMenu/File/Quit by making sure the cleanup is performed by the python MainThread. But at the cost of using SIGUSR1. I think this issue is too trivial for such an expense. Note that without that changeset, the remaining changes fix the leak for ^D, quit(), and h.quit() executed from the terminal.

Reply:
A solution has been developed using SIGTERM instead of SIGUSR1. But I still don't think it is worth it.

Reply:
Are there any additional downsides to reverting said commit? If not, you can revert it and I'll take a final look then.

Reply:
I think there is a greater chance of breakage with the use of SIGUSR1 than  reverting the commit. The thread leak from NEURONMainMenu/File/Quit seems fairly minor to me.